groups/bsl/bslstl/bslstl_variant.h

```diff
@@ -0,0 +1,3338 @@
+    DEST,
+    SOURCE,
+    bsl::void_t<decltype(
+        Variant_ArrayHelper<DEST>{{std::declval<SOURCE>()}})> >
+: bsl::conditional<
+      !Variant_CheckForP1957R1<bool>::value &&
+          bsl::is_same<bool, typename bsl::remove_cvref<DEST>::type>::value &&
+          !bsl::is_same<bool, typename bsl::remove_cvref<SOURCE>::type>::value,
+      bsl::false_type,
+      bsl::true_type>::type {
+    // This metafunction returns true if SOURCE can be converted to DEST
+    // without narrowing.  Before P1957, conversion from pointer or pointer to
+    // member type to bool was not considered a narrowing conversion.  In order
+    // to prevent conversion from pointer or pointer to member type to bool, we
+    // require that 'SOURCE' is 'bool' when 'DEST' is bool.  This means
+    // used-defined alternative types which can be converted to bool may not be
+    // used for construction of a bool alternative.  This is not considered a
+    // problem as current users of 'bdlb::variant' do not rely on conversion
+    // from the initializer, i.e. the DEST and SOURCE types are required to be
+    // identical.
+};
+
+                       // =============================
+                       // struct Variant_OverloadSetImp
+                       // =============================
+/// TODO : this feature (i.e. constructor
+// that doesn't take an index of the alternative to construct) needs expression
+// SFINAE, decltype and declval the implementation takes P0608 into account,
+// which also relies on P1957R2.  In C++03 we only allow construction from an
+// alternative type.  To find the best alternative to construct, we create a
+// class hierarchy 'Variant_OverloadSetImp' that contains a static function
+// 'candidate' for every alternative which can be constructed from source type
+// 'SRC' without narrowing.  The return type of each candidate is set to the
+// 'INDEX'of the respective alternative and is what tells us which alternative
+// to create after applying overload resolution to our set of 'candidate'
+// functions.
+template <class SRC, size_t INDEX, class HEAD, class... TAIL>
+struct Variant_OverloadSetImp
+: Variant_OverloadSetImp<SRC, INDEX + 1, TAIL...> {
+    using Variant_OverloadSetImp<SRC, INDEX + 1, TAIL...>::candidate;
+
+    template <class DEST = HEAD>
+    static typename bsl::enable_if<
+        Variant_ConvertsWithoutNarrowing<DEST, SRC>::value,
+        bsl::integral_constant<size_t, INDEX> >::type candidate(HEAD);
+};
+template <class SRC, size_t INDEX, class HEAD>
+struct Variant_OverloadSetImp<SRC, INDEX, HEAD> {
+    template <class DEST = HEAD>
+    static typename bsl::enable_if<
+        Variant_ConvertsWithoutNarrowing<DEST, SRC>::value,
+        bsl::integral_constant<size_t, INDEX> >::type candidate(HEAD);
+};
+
+template <class SRC, class VARIANT, class = void>
+struct Variant_OverLoadHelper {
+    // Primary template is for cases when there are no candidate alternatives.
```

Comment:
changed to 
```
    // Primary template is for cases when there are no candidate alternatives
    // or in case of an ambiguous choice of alternative type."
```